gen/vsup_zimbra

```diff
@@ -142,43 +153,19 @@ foreach my $rData ($data->getChildElements) {
 		$users->{$uco}->{"LAST_NAME"} = ($lastName || '');
 		$users->{$uco}->{"DISPLAY_NAME"} = ($displayName || '');
 
-		if (defined $uAttributes{$A_BLACKLISTED} and ($uAttributes{$A_BLACKLISTED} == 1)) {
+		if (defined $blacklisted and ($blacklisted == 1)) {
 			# blacklisted users !security ban! are locked
 			$users->{$uco}->{"STATUS"} = 'locked';
 		} else {
-			$users->{$uco}->{"STATUS"} = getStatus($uAttributes{$A_EXPIRATION_KOS},$uAttributes{$A_EXPIRATION_DC2},$uAttributes{$A_EXPIRATION_MANUAL});
+			$users->{$uco}->{"STATUS"} = getStatus($expirationKOS, $expirationDC2, $expirationManual);
 		}
 
+		# manually get mail aliases
+		$users->{$uco}->{$A_EMAIL_VSUP_ALIAS} = $vsupMailAlias || '';
+		$users->{$uco}->{$A_EMAIL_VSUP_ALIASES} = $vsupMailAliases ? join(",", @{$vsupMailAliases}) : '';
 	}
 }
 
-# manually get mail aliases
-my @uco_keys = sort keys %{$users};
-for my $uco (@uco_keys) {
-
-	my $uid = $users->{$uco}->{$A_USER_ID};
-	my @attrNames = ($A_EMAIL_VSUP_ALIAS, $A_EMAIL_VSUP_ALIASES);
-
-	my @attributes = $attributesAgent->getAttributes( user => $uid , attrNames => \@attrNames );
-	foreach (@attributes) {
-		if ($_->getName() eq $A_EMAIL_VSUP_ALIAS) {
-
-			$users->{$uco}->{$A_EMAIL_VSUP_ALIAS} = $_->getValue || '';
-
-		} elsif ($_->getName() eq $A_EMAIL_VSUP_ALIASES) {
-
-			my $aliases = $_->getValue;
-			my @aliases = ();
-			if ($aliases) {
-				@aliases = @$aliases;
-			}
-			$users->{$uco}->{$A_EMAIL_VSUP_ALIASES} = join(",",@aliases) || '';
-
-		}
-	}
-
-}
-
```

Comment:
We must keep manual retrieval of those 2 attributes from the RPC, since we don't want the service to require those attributes (generate them on perun side), but we must consistently propagate them with the rest of params to the Zimbra using this service.

Reply:
Fixed